go/pkg/tree/tree.go

```diff
@@ -108,11 +108,9 @@ func ComputeMerkleTree(execRoot string, is *command.InputSpec, chunkSize int, ca
 }
 
 func buildTree(files map[string]*chunker.Chunker) *treeNode {
-	// This is not the fastest way to build a Merkle tree, but it should do for the intended uses of
-	// this library.
 	root := &treeNode{}
 	for name, ch := range files {
-		segs := strings.Split(name, "/")
+		segs := strings.Split(filepath.Clean(name), string(filepath.Separator))
```

Comment:
nit: use filepath.Split

Reply:
I didn't want to, actually, because it splits into only parts and even then it leaves the separator in, which I don't want.

Reply:
Ah right, you're using all the segs.